worker/src/email/resources/sql/update-stats-email.sql

```diff
@@ -7,7 +7,7 @@ BEGIN
 WITH stats AS (
   SELECT
     COUNT(*) FILTER (WHERE delivered_at IS NULL) AS unsent,
-    COUNT(*) FILTER (WHERE error_code IS NOT NULL) AS errored,
+    COUNT(*) FILTER (WHERE error_code IS NOT NULL AND message_id IS NULL) AS errored,
```

Comment:
Could you add a comment to this to explain the scenario where both of them can be not null? I.e. when a message is processed by the messaging provider but the callback afterwards indicates failure to send